src/libraries/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.Unix.cs

```diff
@@ -173,8 +173,15 @@ private static FileStream CreateSharedBackingObject(Interop.Sys.MemoryMappedProt
         private static FileStream? CreateSharedBackingObjectUsingMemory(
            Interop.Sys.MemoryMappedProtections protections, long capacity, HandleInheritability inheritability)
         {
-            // The POSIX shared memory object name must begin with '/'.  After that we just want something short and unique.
-            string mapName = string.Create(null, stackalloc char[128], $"/corefx_map_{Guid.NewGuid():N}");
+            // The POSIX shared memory object name must begin with '/'.  After that we just want something short (32) and unique.
+            string mapName = Guid.NewGuid().ToString("N");
+            unsafe
+            {
+                fixed (char* mutable = mapName)
+                {
+                    mutable[0] = '/';
```

Comment:
Mutating Strings is not acceptable outside CoreLib.

Reply:
@jkotas thanks for the feedback, I've removed the mutation